src/hooks/useAuth.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+export interface User {
+  email: string;
+  password: string;
+  remember: boolean;
+}
+
+export const useAuth = () => {
+  const [user, setUser] = useState<User>({} as User);
+  const [loading, setLoading] = useState<boolean>(true);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const isUser: User | null = JSON.parse(
```

Comment:
wrong naming. var name starting from `is` means it's a boolean, but you have User | null. Rename please